### PR TITLE
Add 3.12 to tox

### PR DESCRIPTION
**Pull request by jeremydvoss**
# Description

Add tox 3.12 support

# Does This PR Require a Core Repo Change?

https://github.com/open-telemetry/opentelemetry-python/pull/3616

### CODE REVIEW

**Comment by jeremydvoss** [comment]
https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1415

File: .github/workflows/instrumentations_1.yml [hunk 1]
```diff
@@ -39,6 +40,8 @@ jobs:
         exclude:
           - python-version: py311
             package: "prometheus-remote-write"
+          # - python-version: py312
+          #   package: "prometheus-remote-write"
```
Comment (jeremydvoss) [inline .github/workflows/instrumentations_1.yml:44]: Unsure about this "exclude section"

**Comment by jeremydvoss** [comment]
Boto failing. Maybe it's related to this. Maybe we need to update: https://github.com/boto/botocore/issues/2979

**Comment by jeremydvoss** [comment]
Failures:
boto 3.12 20.04: https://github.com/open-telemetry/opentelemetry-python-contrib/actions/runs/9117317763/job/25067996352?pr=2116
botocore 3.12 20.04: https://github.com/open-telemetry/opentelemetry-python-contrib/actions/runs/9117317763/job/25067996921?pr=2116
Kafka-python: https://github.com/open-telemetry/opentelemetry-python-contrib/actions/runs/9117317763/job/25068000409?pr=2116
sio-pika: https://github.com/open-telemetry/opentelemetry-python-contrib/actions/runs/9117317763/job/25068001492?pr=2116
sklearn: https://github.com/open-telemetry/opentelemetry-python-contrib/actions/runs/9117317763/job/25068004112?pr=2116

File: .github/workflows/instrumentations_0.yml [hunk 2]
```diff
@@ -79,6 +80,8 @@ jobs:
           - python-version: py310
             package: "sklearn"
           - python-version: py311
+          #   package: "sklearn"
+          # - python-version: py312
```
[the remark this answers is not in the document]
Comment (xrmx) [reply .github/workflows/instrumentations_0.yml:84]: This is fine, tests fails bump scikit-learn to 1.0.x that is still an old version

File: .github/workflows/instrumentations_1.yml (hunk 1 above)
Comment (xrmx) [reply .github/workflows/instrumentations_1.yml:44]: the test works fine here with python 3.12. Plenty of warnings but tox is happy

```
===================================================== 27 passed, 345 warnings in 0.07s =====================================================
  py312-test-exporter-prometheus-remote-write: OK (17.43=setup[1.48]+cmd[3.12,2.86,3.18,2.97,3.47,0.35] seconds)
```

**Review by xrmx — COMMENTED** [review]
I've added to fix 2 of the failures here https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2529, for the others failures I've added comments

**Comment by jeremydvoss** [comment]
closing in favor or @xrmx's https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2572

**Pull request closed without merging**